conformance/lang/expressions/literals/floating_point_literal.balt

```diff
@@ -0,0 +1,6202 @@
+
+    c = 0X0.0;
+    io:println(c); // @output 0.0
+
+    c = 0X9.9;
+    io:println(c); // @output 9.5625
+
+    c = 0X206.50743;
+    io:println(c); // @output 518.3142728805542
+
+    c = 0Xa.a;
+    io:println(c); // @output 10.625
+
+    c = 0XA.A;
+    io:println(c); // @output 10.625
+
+    c = 0XA.a;
+    io:println(c); // @output 10.625
+
+    c = 0Xa.A;
+    io:println(c); // @output 10.625
+
+    c = 0XaEbcD.0EaD;
+    io:println(c); // @output 715725.0573272705
+
+    c = 0XaE04bcD.0E91aD;
+    io:println(c); // @output 1.8247162905691034E8
+
+    c = 0X.0;
+    io:println(c); // @output 0.0
+
+    c = 0X.1;
+    io:println(c); // @output 0.0625
+
+    c = 0X.2345;
+    io:println(c); // @output 0.1377716064453125
+
+    c = 0X.A;
+    io:println(c); // @output 0.625
+
+    c = 0X.a;
+    io:println(c); // @output 0.625
+
+    c = 0x.aBdECFF;
+    io:println(c); // @output 0.671368595212698
+
+    c = 0x.aBd0E12C9FF4;
+    io:println(c); // @output 0.6711559995600993
+
+    c = 0X.2p+0;
+    io:println(c); // @output 0.125
+
+    c = 0X.AP-0;
+    io:println(c); // @output 0.625
+
+    c = 0X.ap-0;
+    io:println(c); // @output 0.625
+
+    c = 0x.263567p+46;
+    io:println(c); // @output 1.0502700793856E13
+
+    c = 0X.AEDP-46;
+    io:println(c); // @output 9.704043124614259E-15
+
+    c = 0X.aedp-46;
+    io:println(c); // @output 9.704043124614259E-15
+
+    c = 0x.1BeAP+948;
+    io:println(c); // @output 2.594292682612851E284
+
+    c = 0X.0EfCap-456;
+    io:println(c); // @output 3.1462367457512917E-139
+
+    c = 0x0.0p-0;
+    io:println(c); // @output 0.0
+
+    c = 0x0.0P+0;
+    io:println(c); // @output 0.0
+
+    c = 0x9.4p-1;
+    io:println(c); // @output 4.625
+
+    c = 0x9546.4748p+126;
+    io:println(c); // @output 3.250911279637454E42
+
+    c = 0xa.aP-1;
+    io:println(c); // @output 5.3125
+
+    c = 0Xa.ap+1;
+    io:println(c); // @output 21.25
+
+    c = 0xA.aP-12;
+    io:println(c); // @output 0.002593994140625
+
+    c = 0XA.ap+12;
+    io:println(c); // @output 43520.0
+
+    c = 0xaeb.acdfP-302;
+    io:println(c); // @output 3.431057809225094E-88
+
+    c = 0xABC.CEEFFp+302;
+    io:println(c); // @output 2.2397685993041007E94
+
+    c = 0xAbC.0eEfFp-453;
+    io:println(c); // @output 1.1815113934486016E-133
+
+    c = 0xAb12C.0e9E1f0FP+45;
+    io:println(c); // @output 2.465425448163409E19
+}
+
+Test-Case: error
+Description: Test hex floating point literal with decimal type descriptor.
+Labels: decimal, HexFloatingPointLiteral
+
+function errorFunction() {
+    decimal _ = 0x1.0fp+10; // @error hex floating point literal is incompatible  with 'decimal'
+
+    decimal _ = 0x1p-5; // @error hex floating point literal is incompatible  with 'decimal'
+
+    decimal _ = 0x1.0; // @error hex floating point literal is incompatible  with 'decimal'
+}
+
+Test-Case: output
+Description: Test hex floating point literal type narrowing with floating point type descriptor.
```

Comment:
Type narrowing is on the variable, not the literal. Can we update this to say narrowing when a literal is used with the union of float and decimal instead?